src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -517,13 +517,14 @@ private HttpClient CreateHttpClient()
                 {
                     httpClientHandler.Proxy = _httpConnectionOptions.Proxy;
                 }
-                if (_httpConnectionOptions.Cookies != null)
+
+                // Only access HttpClientHandler.ClientCertificates and HttpClientHandler.CookieContainer
+                // if the user has configured client certs
```

Comment:
```suggestion
                // if the user has configured those options
```

Since it's not just certs now ;)